xarray_beam/_src/core.py

```diff
@@ -287,18 +295,19 @@ def __init__(
         rather than only on the host process. This is important for scaling
         pipelines to millions of tasks.
     """
+    self.dataset = dataset
+    self._validate(dataset, split_vars)
     if chunks is None:
-      chunks = dataset.chunks
+      chunks = self._first.chunks
     if chunks is None:
-      raise ValueError('dataset must be chunked or chunks must be provided')
-    expanded_chunks = normalize_expanded_chunks(chunks, dataset.sizes)
-    self.dataset = dataset
+      raise ValueError("dataset must be chunked or chunks must be provided")
+    expanded_chunks = normalize_expanded_chunks(chunks, self._first.sizes)
     self.expanded_chunks = expanded_chunks
     self.split_vars = split_vars
     self.num_threads = num_threads
     self.shard_keys_threshold = shard_keys_threshold
     # TODO(shoyer): consider recalculating these potentially large properties on
-    # each worker, rather than only once on the host.
+    #  each worker, rather than only once on the host.
```

Comment:
remove extra leading space

Reply:
Mind if I keep this space? I intended to add this in order to make highlighting work in my IDE. 

https://user-images.githubusercontent.com/2322439/214760990-77596929-2397-4d12-8482-c749af0be10a.mov

Reply:
I'm not really opposed, but I haven't seen this convention for indenting TODO note before and it isn't used elsewhere in this codebase (or google3 for that matter). I'd rather not do this as a one-off change.

VS Code just highlights the word "TODO" in a bright color instead.